File: packages/sdui/src/main/java/nl/q42/sdui/screen/SettingsScreen.java
package nl.q42.sdui.screen;

import org.springframework.stereotype.Component;

import nl.q42.common.screen.Screen;

@Component
public class SettingsScreen extends Screen {

    public static final String SCREEN_NAME = "settings";

    public SettingsScreen() {
        super(SCREEN_NAME);
    }
}

File: packages/sdui/src/main/kotlin/nl/q42/sdui/screen/SettingsScreen.kt
package nl.q42.sdui.screen

import nl.q42.common.ServerComponent
import nl.q42.common.TextComponent
import nl.q42.core.AppRequestContext

@org.springframework.stereotype.Component
class SettingsScreen : ScreenInstance {

    override fun content(context: AppRequestContext): List<ServerComponent> {
        return listOf(
            TextComponent(text = "Settings", contentId = "settings-text-1")
        )
    }

    override fun name(): String {
        return SCREEN_IDENTIFIER
    }

    companion object {
        const val SCREEN_IDENTIFIER: String = "settings"
    }
}
